protocol/consistencychecks_test.go

```diff
@@ -46,6 +46,20 @@ func TestVerifyWithError(t *testing.T) {
 	}
 }
 
+func TestMalformedClientMessage(t *testing.T) {
+	d, pk := NewTestDirectory(t, true)
+	cc := NewCC(d.LatestSTR(), true, pk)
+
+	request := &RegistrationRequest{
+		Username: "", // invalid username
+		Key:      key,
+	}
+	res, _ := d.Register(request)
+	if err := cc.HandleResponse(RegistrationType, res, "", key); err != ErrMalformedClientMessage {
+		t.Error("Unexpected verification result")
+	}
+}
```

Comment:
I think currently we have no limitations for the key size or name length? And this test is mostly for `HandleResponse` with `msg.validate()`. Other tests could be (and already) covered in `directory_test.go`.